raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -63,6 +63,7 @@ public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IOException {
             store.clear();
 
             election = ElectionState.withUnknownLeader(0);
+            state = new FollowerState(election.epoch);
```

Comment:
Hmm.. Shouldn't this go under `election.epoch < logEndOffsetAndEpoch.epoch`? In every other branch below, we initialize `state`.

Reply:
We have the logic to look up epoch from state in 
```
 private boolean becomeFollower(int newEpoch, Function<FollowerState, Boolean> func) throws IOException {
        int currentEpoch = epoch();
```
which means we need to initialize the state first.